src/store.c

```diff
@@ -106,8 +106,10 @@ static void garbage_collect_dead_processes(struct ps_struct *ps_first) {
                         /* close the stream and fds */
                         ps_next->schedstat = safe_close(ps_next->schedstat);
                         ps_next->sched = safe_close(ps_next->sched);
-                        if (ps->smaps)
-                                fclose(ps->smaps);
+                        if (ps_next->smaps) {
+                                fclose(ps_next->smaps);
+                                ps_next->smaps = NULL;
```

Comment:
I don't think you need to set this to NULL. I think the reason for the double free was that it was using ps instead of ps_next. I won't hurt to set it to NULL though.